### DISCUSSION TITLE
[DEV] Keira3 and different wow versions/emulators

### DISCUSSION THREAD

**Discussion by FrancescoBorzi** [azerothcore/azerothcore-wotlk#3920] [General]
This topic has been discussed already but I want to give it extra visibility and give an official explaination.

Frequent asked questions: **Does Keira3 support other WoW versions/emulators?**

The answer is **Keira3 is tailored for AzerothCore**

## What does it mean in practice?

- Keira3 features might or might not work with other emulators, it mainly depends on the **database structure**.

- Other wow emulators are not supported. Bug reports for other emus will not be accepted.

## Why?

What breaks a DB tool when using it on a different emulator (or WoW version) is the difference in the **DB structure**.

In the old Keira2 we have tried to support both TC `3.3.5` and `master` branch and that was quite unhandy: it added extra complexity to the tool application.

I don't believe this extra complexity should be handled in the editor tool.

I believe that the tables that represents the same things (creatures, quests, gameobjects, etc...), **simply should have the same structure**. Unfortunately that is not always the case, but:

## You can help with that

In past, I have spent a lot of time and effort trying to align the TC `3.3.5` DB to TC `master` branch (see my commits in TC 3.3.5 branch of some years ago)  as well as trying to align AC with TC (see this: https://github.com/azerothcore/azerothcore-wotlk/issues/1118).

I believe it makes more sense to spend time in doing this, so automatically all tools will be compatible with all emulators and wow versions.

I personally stopped in doing such activities since I can't afford it, I don't have time for it.

But **you** can help with it.

So instead of demanding tools to add complexity in their code (and spend time) to support different structures, you can spend your own time in aligning the DB structures. That would benefit, not only the usage of Keira3, but even the usage of other tools as well as gaining the ability to easily import DB stuff from one project to another.

**Comment by BarbzYHOOL** [comment]
Might put this in a doc file (.md) in Keira repo, what do you think?

**Comment by Quest79** [comment]
I agree with your philosophy about DB structure. Although I understand some structure may be different between 3.3.5 and 9.x+ due to additions, I hope that TC and AC (and others) will unify structure to a single form. That is best practice for the future of Wow emulation for decades to come.
